### PR TITLE
Include missing data count in pd.DataFrame.describe()

### LINKED ISSUE #21689: Include missing data count in pd.Dataframe.describe method
#### Code Sample

```python
d = {'col1': [1, np.nan], 'col2': [3, 4]}
df = pd.DataFrame(data=d)
df.describe()
```
#### Problem description
* Output
image

The describe method generally only include 9 summary statistics(count, mean, std, min, 25%, 50%, 75%, max, missing) but no **missing count** which is very import in realworld data analysis.

To include missing count I have to use the following code,
```python
d = {'col1': [1, np.nan], 'col2': [3, 4]}
df = pd.DataFrame(data=d)
des1 = df.describe()
des2 = df.isnull().sum().to_frame(name = 'missing').T
pd.concat([des1, des2])
```
And the output
image

#### Expected Output
Expect include missing count in describe method.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.5.final.0
python-bits: 64
OS: Darwin
OS-release: 16.7.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: en_US.UTF-8
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.23.0
pytest: 3.5.1
pip: 10.0.1
setuptools: 39.1.0
Cython: 0.28.2
numpy: 1.14.3
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 6.4.0
sphinx: 1.7.4
patsy: 0.5.0
dateutil: 2.7.3
pytz: 2018.4
blosc: None
bottleneck: 1.2.1
tables: 3.4.3
numexpr: 2.6.5
feather: None
matplotlib: 2.2.2
openpyxl: 2.5.3
xlrd: 1.1.0
xlwt: 1.2.0
xlsxwriter: 1.0.4
lxml: 4.2.1
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.2.7
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by alexander-ponomaroff**
- [x] closes #21689
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by alexander-ponomaroff** [comment]
I would like to receive some feedback on the added summary statistic to `describe()`. It's between having `missing`, which counts the number of missing values, and having `length`, which will include missing values and if the user wants to get the missing summary, they will have to subtract `count` from `length`. I bring this up because @jreback asked for `length` in the issue, but the creator of the issue asked for `missing`. In my opinion, I think that the way I currently did it with `missing` is better.

Please let me know what everyone thinks, and once that's out of the way, I will proceed to change the documentation in the function to reflect the new enhancement.

**Comment by WillAyd** [comment]
Missing is not a method that exists in our API so I don't think it should be added here. Size seems more appropriate to me

**Review by WillAyd — CHANGES_REQUESTED** [review]
Can you merge master and address comment?

**Comment by alexander-ponomaroff** [comment]
@WillAyd Thanks for the feedback, let me know if you agree with the change now that I changed 'missing' to 'size'. If all good, I will go ahead and fix other tests that are failing due to the added functionality.

File: pandas/core/generic.py [hunk 1]
```diff
@@ -9804,9 +9804,10 @@ def describe(self, percentiles=None, include=None, exclude=None):
 
         def describe_numeric_1d(series):
             stat_index = (['count', 'mean', 'std', 'min'] +
-                          formatted_percentiles + ['max'])
+                          formatted_percentiles + ['max', 'size'])
             d = ([series.count(), series.mean(), series.std(), series.min()] +
```
Comment (jreback) [inline pandas/core/generic.py]: should be the first arg

File: doc/source/whatsnew/v0.25.0.rst [hunk 2]
```diff
@@ -420,7 +420,7 @@ Other
 ^^^^^
 
 - Removed unused C functions from vendored UltraJSON implementation (:issue:`26198`)
-
+- Added enhancement to :func:`pd.DataFrame.describe` to include size as one of the summary statistics (:issue:`21689`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: this needs a separate subsection, its actually a fairly large API change.

**Review by jreback — CHANGES_REQUESTED** [review]
I am only +0 on adding this.

**Comment by jreback** [comment]
can you update per comments

**Comment by alexander-ponomaroff** [comment]
Sorry for the long delay, I was away on vacation. I will be making the updates within the next couple of days.

**Comment by jreback** [comment]
closing as stale, ping if you'd like to continue.

**Comment by drkarthi** [comment]
@jreback @WillAyd I am interested in having the number of rows in the dataframe as an output in describe(). I assume that is what is being called `size` here. Is there still interest to have this added? I am happy to pick up from where @alexander-ponomaroff left off

**Comment by drkarthi** [comment]
@jbrockmendel @mroeschke I see you are active contributors right now. do you you have any objections to adding the number of rows in the dataframe as an output of describe()

**Comment by jbrockmendel** [comment]
i have no opinion on describe.  if mroeschke expresses an opinion ill agree with that, otherwise ill agree with jreback at +0.

**Pull request closed without merging**